warehouse/query-core/src/main/java/datawave/query/planner/DefaultQueryPlanner.java

```diff
@@ -1236,22 +1243,34 @@ protected ASTJexlScript processTree(final ASTJexlScript originalQueryTree, Shard
             
             // Expand any bounded ranges into a conjunction of discrete terms
             try {
-                innerStopwatch = timers.newStartedStopwatch("DefaultQueryPlanner - Expand regex");
                 ParallelIndexExpansion regexExpansion = new ParallelIndexExpansion(config, scannerFactory, metadataHelper, expansionFields,
                                 config.isExpandFields(), config.isExpandValues(), config.isExpandUnfieldedNegations());
-                queryTree = (ASTJexlScript) regexExpansion.visit(queryTree, null);
-                if (log.isDebugEnabled()) {
-                    logQuery(queryTree, "Query after expanding regex:");
+                
+                // Check if there is any regex to expand.
+                NodeTypeCount nodeCount = NodeTypeCountVisitor.countNodes(queryTree);
+                if (nodeCount.hasAny(ASTNRNode.class, ASTERNode.class)) {
+                    innerStopwatch = timers.newStartedStopwatch("DefaultQueryPlanner - Expand regex");
+                    queryTree = (ASTJexlScript) regexExpansion.visit(queryTree, null);
+                    if (log.isDebugEnabled()) {
+                        logQuery(queryTree, "Query after expanding regex:");
+                    }
+                    innerStopwatch.stop();
+                    
+                    // The query tree was modified. Recount nodes.
+                    nodeCount = NodeTypeCountVisitor.countNodes(queryTree);
```

Comment:
I don't think we need to do this here.  Just do it once after expanding regexes and ranges.

Reply:
Removed.